src/main/java/com/example/demo/service/CommentaryService.java

```diff
@@ -0,0 +1,41 @@
+package com.example.demo.service;
+
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.example.demo.entity.Commentary;
+import com.example.demo.repository.CommentaryRepository;
+
+
+@Service
+public class CommentaryService {
+    
+
+    @Autowired
+    private CommentaryRepository commentaryRepository;
+
+    public List<Commentary> getAll() {
+        List<Commentary> commentaries = this.commentaryRepository.findAll();
+        return commentaries;
+    }
+
+    public Commentary getOneById(Long id) {
+        Commentary commentary = this.commentaryRepository.findById(id).get();
+        return commentary;
+    }
+
+    public Commentary getOneBySlug(String slug) {
+        Commentary commentary = this.commentaryRepository.findBySlug(slug);
+        return commentary;
+    }
+
+    public Commentary createCommentary(Commentary commentary) {
+        Commentary createdCommentary = this.commentaryRepository.save(commentary);
```

Comment:
commentaries have title ? if not it will be a bit long url if you retrieve it by slug ? 

In any case, as long as you create a slug you have to "slugify" the title / content you want to use in a url